api/base/filters.py

```diff
@@ -61,6 +66,83 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class NewDjangoFilterMixin(django_filters.FilterSet):
+
+    QUERY_PATTERN = re.compile(r'^filter\[(?P<fields>((?:,*\s*\w+)*))\](\[(?P<op>\w+)\])?$')
+    FILTER_FIELDS = re.compile(r'(?:,*\s*(\w+)+)')
+
+    or_fields = []
+
+    def __init__(self, data=None, *args, **kwargs):
+        if data:
+            new_qd = QueryDict('', mutable=True)
+            for key, value in data.iteritems():
+                match = self.QUERY_PATTERN.match(key)
+                if match:
+                    match_dict = match.groupdict()
+                    fields = match_dict['fields']
+                    field_names = re.findall(self.FILTER_FIELDS, fields.strip())
+
+                    self.or_fields = []
+                    multiple_values = value.split(',')
+                    if len(multiple_values) > 1 or len(field_names) > 1:
```

Comment:
I had a whole lot of trouble getting this to work out of the box -- theoretically, from reading through a few issues and their resolutions, the only mention I can find on how to handle those natively was through the [CSV Widget] -- http://django-filter.readthedocs.io/en/develop/ref/widgets.html#csv-widget -- though applying that in my testing of things didn't get me anywhere for a long while so I decided to just deal with it while dealing with other `OR` cases at the same time. I can dive back in tho and really try to figure out what's happening....